src/facade/dragonfly_connection.cc

```diff
@@ -769,6 +785,26 @@ auto Connection::IoLoop(util::FiberSocketBase* peer, SinkReplyBuilder* orig_buil
   do {
     FetchBuilderStats(stats_, orig_builder);
 
+    if (migration_request_ && cc_->subscriptions == 0) {
+      ProactorBase* dest = migration_request_;
+
+      if (dispatch_fb_.IsJoinable()) {
+        SendAsync({MigrationRequestMessage{}});
+        dispatch_fb_.Join();
+      }
+
+      migration_request_ = nullptr;
+      this->Migrate(dest);
+
+      // We're now running in `dest` thread
+      queue_backpressure_ = &tl_queue_backpressure_;
+      // In case we Yield()ed in Migrate() above, dispatch_fb_ might have been started.
+      if (!dispatch_fb_.IsJoinable()) {
+        dispatch_fb_ = fb2::Fiber(dfly::Launch::post, "connection_dispatch",
+                                  [&, peer] { DispatchFiber(peer); });
+      }
```

Comment:
I'd introduce some kind of function: launch-if-needed that will be called here and in SendAsync